### PR TITLE
Created an option to find modules via identifiers

### PR DESCRIPTION
**Pull request by saxenism**
As a fix for the issue mentioned by the auditors in issue #317 regarding the `findModuleAddressInOrchestrator` not designed to handle the cases where multiple modules with the same name are being used, I have made a few changes to `findModuleAddressInOrchestrator`.

Since the URL of a module is much less likely to be the same as other modules, I have replaced the query string from the module title to the module URL.

And since, it is technically possible to have multiple modules with the same URL, i have also given an option to search for module address using a **unique module identifier**.

### CODE REVIEW

**Comment by FHieser** [comment]
Where tests

File: src/orchestrator/Orchestrator.sol
Comment (FHieser) [inline src/orchestrator/Orchestrator.sol:1]: The tests are missing

**Review by FHieser — CHANGES_REQUESTED** [review]

File: src/orchestrator/Orchestrator.sol
Comment (saxenism) [reply src/orchestrator/Orchestrator.sol:1]: image

The tests are no longer missing 👉👈

**Comment by saxenism** [comment]
Considering this comment:


Added the documentation for `dependencyData`.

The `dependencyData` now expects 3 fields:
1. bool hasDependency: True if dependencies are expected, false otherwise
2. string[] dependenciesURLs: URL of the modules that can be counted as dependencies
3. bytes additionalData: Any additional data that you might want access to, inside of `init2`.

**Comment by FHieser** [comment]
The changes will be implemented with the #371 PR

**Pull request closed without merging**